### PR TITLE
Adding support for version 2023-09-01

### PR DESCRIPTION
**Pull request by osid29**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by dolauli** [comment]
@osid29 According to the CI failures, there are some issues in your examples. Please fix them.

**Comment by osamasid** [comment]
@dolauli I have updated the PR and the checks seems to pass now. Please help with the merge

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1 [hunk 1]
```diff
@@ -19,156 +19,156 @@ Describe 'AzPaloAltoNetworks' {
         {
             $config = New-AzPaloAltoNetworksLocalRulestack -Name $env.LocalRulestackName -ResourceGroupName $env.resourceGroup -Location $env.location -Description "testing powershell" -DefaultMode 'NONE'
             $config.Name | Should -Be $env.LocalRulestackName
-        } | Should -Not -Throw
```
Comment (dolauli) [inline src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1:22]: Is special reason that you remove `Should -Not -Throw` Here.

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/Get-AzPaloAltoNetworksFirewall.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  $currentPath = $PSScriptRoot
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Get-AzPaloAltoNetworksFirewall' {
+    It 'List' -skip {
```
Comment (dolauli) [inline src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/Get-AzPaloAltoNetworksFirewall.Tests.ps1:18]: Please add tests for those new added cmdlets.

**Review by dolauli — CHANGES_REQUESTED** [review]

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1 (hunk 1 above)
Comment (osid29) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1:22]: The build was failing due to missing exception

https://dev.azure.com/azclitools/public/_build/results?buildId=110806&view=logs&j=b17395f6-68a3-5682-0476-d3f6f1043109&t=e50c9eb1-7f19-5aa6-670b-cc46e51b352e

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/Get-AzPaloAltoNetworksFirewall.Tests.ps1 (hunk 2 above)
Comment (osid29) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/Get-AzPaloAltoNetworksFirewall.Tests.ps1:18]: There are no newly added cmdlets, correct me please if you see any.

Comment (osamasiddiqui29) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/Get-AzPaloAltoNetworksFirewall.Tests.ps1]: @dolauli I figured out that these tests are not skipped but are duplicates. All cmdlets scenarios are already covered in **AzPaloAltoNetworks.Tests.ps1** in create, get, update, remove fashion. I have updated this PR 
image

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1 (hunk 1 above)
Comment (dolauli) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1:22]: > Is special reason that you remove `Should -Not -Throw` Here.

@osamasid would please revert the change to this file?

**Comment by dolauli** [comment]
@osid29 Any comments to my first comment

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1 (hunk 1 above)
Comment (osid29) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/test/AzPaloAltoNetworks.Tests.ps1]: I reverted the changes and the CI checks are passing now

**Comment by osid29** [comment]
> @osid29 Any comments to my first comment

@dolauli I have addressed both the comments. The CI checks are passing as well

**Review by dolauli — APPROVED** [review]

File: src/PaloAltoNetworks/PaloAltoNetworks.Autorest/docs/Az.PaloAltoNetworks.md [hunk 3]
```diff
@@ -107,12 +107,24 @@ Reset counters
 ### [Save-AzPaloAltoNetworksFirewallLogProfile](Save-AzPaloAltoNetworksFirewallLogProfile.md)
 Log Profile for Firewall
```
Comment (dolauli) [inline src/PaloAltoNetworks/PaloAltoNetworks.Autorest/docs/Az.PaloAltoNetworks.md:109]: I just noticed you do not add examples and tests for four new cmdlets. These four new cmdlets are new feature of autorest.powershell v4. In v4, if there is no patch API for a resource, we will generate an update-xxx cmdlet by using get->put API to simulate a Patch API. 
I would suggest you keep those four cmdlets and add examples and tests for them. If you do not need them, please let me know and I will remove those four cmdlets through a directive.

Comment (osid29) [reply src/PaloAltoNetworks/PaloAltoNetworks.Autorest/docs/Az.PaloAltoNetworks.md:109]: Hi @dolauli, we don't need below cmdlets now. If we see any adoption, we can update these in future requests. Please remove these.

'Update-AzPaloAltoNetworksFqdnListLocalRulestack', 
'Update-AzPaloAltoNetworksLocalRule',
'Update-AzPaloAltoNetworksCertificateObjectLocalRulestack'
'Update-AzPaloAltoNetworksPrefixListLocalRulestack'

**Pull request merged**